repo/blob/gcs/gcs_storage_test.go

```diff
@@ -23,35 +30,37 @@ func TestGCSStorage(t *testing.T) {
 		t.Skip("skipping test because GCS credentials file can't be opened")
 	}
 
-	ctx := testlogging.Context(t)
-	st, err := gcs.New(ctx, &gcs.Options{
-		BucketName:                   bucket,
-		ServiceAccountCredentialJSON: credData,
-	})
+	testlease.RunWithLease(t, testLeaseID, testLeaseDuration, func() {
+		ctx := testlogging.Context(t)
+		st, err := gcs.New(ctx, &gcs.Options{
+			BucketName:                   bucket,
```

Comment:
If a prefix is specified, then the tests could share the bucket and not step into each other toes. Would that work?
In fact, we'd want to test when multiple repos share the same bucket.
Same for the S3 tests.